website/blog/2024-06-03-monitor-ferretdb-performance-using-coroot.md

```diff
@@ -0,0 +1,136 @@
+
+```sh
+docker restart <prometheus_container_name>
+```
+
+Comfirm that the FerretDB metrics are being collected by Prometheus by navigating to the Prometheus targets at http://localhost:9090/targets.
+
+![Prometheus targets](/img/blog/ferretdb-coroot/prometheus-targets.png)
+
+Once the setup is complete, Coroot will start collecting metrics from FerretDB.
+
+## Monitor FerretDB performance with Coroot
+
+The Coroot dashboard provides the full details on all components.
+
+Ensure Prometheus integration is correctly set up to collect metrics from FerretDB.
+
+![Coroot prometheus integration](/img/blog/ferretdb-coroot/prometheus-integration.png)
+
+From the Coroot dashboard, you can view the FerretDB metrics through Prometheus.
+
+![CPU dashboard 1](/img/blog/ferretdb-coroot/cpu-metrics-1.png)
+
+![CPU dashboard 1](/img/blog/ferretdb-coroot/cpu-metrics-2.png)
+
+In the images, the FerretDB instance indicates a peak Requests Per Second (RPS) of 0.07 with a consistent 2ms latency.
+
+![memory usage](/img/blog/ferretdb-coroot/memory-metrics.png)
+
+Looking at the memory usage metrics.
+you can see that the system is effectively managing memory resources without any significant performance degradation.
+They show a gradual increase in usage, peaking at around 20MB without any out-of-memory issues or significant leaks.
+
+We can also monitor the Postgres database to know how it interacts and handles all requests from FerretDB.
+
+![Postgres dashboard](/img/blog/ferretdb-coroot/postgres-cpu-1.png)
+
+![Postgres dashboard](/img/blog/ferretdb-coroot/postgres-cpu-2.png)
+
+The Postgres container indicates a sharp CPU usage spike around 18:00, mirrored by a similar increase in CPU delay.
+Despite the spike, throttled time remains negligible.
+Node CPU usage also show moderate fluctuations but stays within acceptable limits.
+So far, we can see that the system can handle peak loads without significant performance loss.
+
+It also demonstrates stable I/O performance with low latency around 0.5ms and consistent I/O utilization peaking at 4%.
+IOPS remains steady at approximately 75-100 and the bandwidth usage stays around 2-3MB/s.
+
+## Analyze and optimize FerretDB performance with Coroot
+
+Coroot provides a comprehensive view of your entire system, including FerretDB and all its components.
+You can monitor and analyze performance metrics in real-time, identify bottlenecks, and optimize your system for better performance.
+It also includes predefined inspections of your system and its components, including SLO error rate/latency and notifications.
+You also get CPU, memory, storage, network, and log management metrics.
+
+To get started with FerretDB, [see our documentation](https://docs.ferretdb.io/).
+And if you want to contact the team for help or have any questions, [contact us on Slack](https://join.slack.com/t/ferretdb/shared_invite/zt-zqe9hj8g-ZcMG3~5Cs5u9uuOPnZB8~A).
```

Comment:
As we talked about many times before, we should not use that link because it may change. We should link to the community section in our docs or in README